extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -784,33 +809,103 @@ public void onFetchPage(@Nonnull final Downloader downloader)
         }
     }
 
+    /**
+     * Fetch the Android Mobile API or fallback to the desktop streams.
+     * If something went wrong when parsing this API, fallback to the desktop JSON player, fetched
+     * again if the {@code signatureTimestamp} of the JS player is unknown (because signatures
+     * without a {@code signatureTimestamp} included in the player request are invalid).
+     */
+    private void fetchAndroidMobileJsonPlayer(final ContentCountry contentCountry,
+                                              final Localization localization,
+                                              final String videoId) throws ExtractionException,
+            IOException {
+        JsonObject mobilePlayerResponse = null;
+        final byte[] mobileBody = JsonWriter.string(prepareMobileJsonBuilder(localization,
+                contentCountry)
+                .value("videoId", videoId)
+                .done())
+                .getBytes(UTF_8);
+        try {
+            mobilePlayerResponse = getJsonMobilePostResponse("player", mobileBody,
+                    contentCountry, localization);
+        } catch (final Exception ignored) {
+        }
+        if (mobilePlayerResponse != null && mobilePlayerResponse.has("streamingData")) {
+            final JsonObject mobileStreamingData = mobilePlayerResponse.getObject(
+                    "streamingData");
+            if (!isNullOrEmpty(mobileStreamingData)) streamingData = mobileStreamingData;
+        } else {
+            // Fallback to the desktop JSON player endpoint
+
+            // The cipher signatures from the player endpoint without a timestamp are invalid so
+            // download it again only if we didn't have a signatureTimestamp before fetching the
+            // data of this video (the sts string).
+            if (sts == null && isCipherProtectedContent()) {
+                getStsFromPlayerJs();
+                final JsonObject playerResponseWithSignatureTimestamp = getJsonPostResponse(
+                        "player", createPlayerBodyWithSts(localization, contentCountry, videoId),
+                        localization);
+                if (playerResponseWithSignatureTimestamp.has("streamingData")) {
+                    streamingData = playerResponseWithSignatureTimestamp.getObject(
+                            "streamingData");
+                }
+            } else {
+                streamingData = playerResponse.getObject("streamingData");
+            }
+        }
```

Comment:
I see that this method sometimes does not assign anything to `streamingData`, is that normal?